### ISSUE TITLE
About loss function 

### ISSUE THREAD

**Issue by YangYL18** [ZihaoWang-CV/CAMP_iccv19#6]
Hello, thanks for sharing your code! I had some doubts.
In your paper, the loss function is ` log(m(I,C))+max[log(1-m(I,C'))]+log(m(I,C))+max[log(1-m(I',C))]` , however,in your code the loss function is `le = -(scores.sum() + scores.sum() + de_scores.min(1)[0].sum() + de_scores.min(0)[0].sum())`，which is correct, max or min? I am confused about it.

Thank you very much!

**Comment by YangYL18** [comment]
Hello, when I use  `config_finetun.yaml` as config file, the recall@k was `Image to Text: 0.1, 0.5, 1.0, 836.0, 1055.4     Tetx to Image: 0.1, 0.5, 0.9, 499.0, 500.7` (I changed the bath_size=64).Can you tell me the reason, I am confused about it.
Thank you very much!

**Comment by ZihaoWang-CV** [comment]
Sorry for my late reply.

For the first question about loss function:
Thanks for the correction! The loss term of the hardest negative example in our paper should be `max[-log(1-m(I,C'))]`, and for the positive example, it should be `-log(m(I,C))` . Sorry for the confusion!


For the question about training results:
We first train the baseline cross-attention model in our experiment, then fine-tuning the gate model on the pre-trained cross-attention model. Please refer to the README.md or our supplementary material for details.

Thanks!

**Comment by haoopan** [comment]
> Hello, when I use `config_finetun.yaml` as config file, the recall@k was `Image to Text: 0.1, 0.5, 1.0, 836.0, 1055.4 Tetx to Image: 0.1, 0.5, 0.9, 499.0, 500.7` (I changed the bath_size=64).Can you tell me the reason, I am confused about it.
> Thank you very much!


I also fcaing the problems, even training the cross-attention model. Do you have any ideas?

**Comment by ZihaoWang-CV** [comment]
> > Hello, when I use `config_finetun.yaml` as config file, the recall@k was `Image to Text: 0.1, 0.5, 1.0, 836.0, 1055.4 Tetx to Image: 0.1, 0.5, 0.9, 499.0, 500.7` (I changed the bath_size=64).Can you tell me the reason, I am confused about it.
> > Thank you very much!
> 
> I also facing the problems, even training the cross-attention model. Do you have any ideas?


Hi. Are you using the config file `experiments/f30k_cross_attention/config.yaml` or `experiments/f30k_cross_attention/config_256.yaml `? 
As far as I know, those two configs are easy to train and reproduce results. Did the loss also not decrease during the training?

**Comment by haoopan** [comment]
> > > Hello, when I use `config_finetun.yaml` as config file, the recall@k was `Image to Text: 0.1, 0.5, 1.0, 836.0, 1055.4 Tetx to Image: 0.1, 0.5, 0.9, 499.0, 500.7` (I changed the bath_size=64).Can you tell me the reason, I am confused about it.
> > > Thank you very much!
> > 
> > 
> > I also facing the problems, even training the cross-attention model. Do you have any ideas?
> 
> Hi. Are you using the config file `experiments/f30k_cross_attention/config.yaml` or `experiments/f30k_cross_attention/config_256.yaml `?
> As far as I know, those two configs are easy to train and reproduce results. Did the loss also not decrease during the training?

Thank you for your answer. Yes, I used config file **experiments/f30k_cross_attention/config_256.yaml**. After more than **200** epoches of training, the loss was always around **177.4**. In the code I just changed **batch_size=128** to **batch_size=64** at the time of training and  **bs =10** to **bs =1** at the time of evaluation.

**Comment by ZihaoWang-CV** [comment]
> > > > Hello, when I use `config_finetun.yaml` as config file, the recall@k was `Image to Text: 0.1, 0.5, 1.0, 836.0, 1055.4 Tetx to Image: 0.1, 0.5, 0.9, 499.0, 500.7` (I changed the bath_size=64).Can you tell me the reason, I am confused about it.
> > > > Thank you very much!
> > > 
> > > 
> > > I also facing the problems, even training the cross-attention model. Do you have any ideas?
> > 
> > 
> > Hi. Are you using the config file `experiments/f30k_cross_attention/config.yaml` or `experiments/f30k_cross_attention/config_256.yaml `?
> > As far as I know, those two configs are easy to train and reproduce results. Did the loss also not decrease during the training?
> 
> Thank you for your answer. Yes, I used config file **experiments/f30k_cross_attention/config_256.yaml**. After more than **200** epoches of training, the loss was always around **177.4**. In the code I just changed **batch_size=128** to **batch_size=64** at the time of training and **bs =10** to **bs =1** at the time of evaluation.

Sorry I did not encounter this issue before. What is the version of your Pytorch?

I may suggest checking the following settings before I could reproduce this issue:

1.  Check the data format. 
2.  I'm not sure whether the usage of LSTM and pack_padded_sequence are changed in an updated version of Pytorch.
3.  Check whether the cross-attention layers cause this issue. I may remove the attention layer, just use the baseline setting (Use average pooling to aggregate image/word features and simple margin loss used in VSE++ to train the LSTM and FC layers. See whether the loss could decrease correctly during the training)

**Comment by SnoopyMark** [comment]
> I also fcaing the problems, even training the cross-attention model. Do you have any ideas?

Same question, have you solved the problem now? thx

**Comment by NJUST-KMG** [comment]
> > > > Hello, when I use `config_finetun.yaml` as config file, the recall@k was `Image to Text: 0.1, 0.5, 1.0, 836.0, 1055.4 Tetx to Image: 0.1, 0.5, 0.9, 499.0, 500.7` (I changed the bath_size=64).Can you tell me the reason, I am confused about it.
> > > > Thank you very much!
> > > 
> > > 
> > > I also facing the problems, even training the cross-attention model. Do you have any ideas?
> > 
> > 
> > Hi. Are you using the config file `experiments/f30k_cross_attention/config.yaml` or `experiments/f30k_cross_attention/config_256.yaml `?
> > As far as I know, those two configs are easy to train and reproduce results. Did the loss also not decrease during the training?
> 
> Thank you for your answer. Yes, I used config file **experiments/f30k_cross_attention/config_256.yaml**. After more than **200** epoches of training, the loss was always around **177.4**. In the code I just changed **batch_size=128** to **batch_size=64** at the time of training and **bs =10** to **bs =1** at the time of evaluation.

Same question, Could you give me some advice? Thanks.

**Comment by ZihaoWang-CV** [comment]
Hi all, sorry for my late reply.
It seems that a PyTorch version later than 1.7 may cause that problem. I'm trying to detect and fix the issue. 
It could be very helpful if you could provide your PyTorch environments.
Thanks all!

**Comment by luna0804** [comment]
> 

I used torch1.4, but seemed to run into the same problem......
